xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -38,6 +38,28 @@ using namespace XFILE;
 
 namespace UPNP
 {
+// add some hard coded allowed subtitle just like
+// https://github.com/xbmc/xbmc/blob/bdf0f9b468475a4fc4e603eebef004175182968c/xbmc/network/upnp/UPnPInternal.cpp#L663 did
+// see also line#684
+// the original version of content type here,eg: text/srt,which was defined 10 years ago (year 2013,
+// https://github.com/xbmc/xbmc/blame/00d752bfbe862c67e189d461859e31f471981d33/xbmc/network/upnp/UPnPInternal.cpp#L1158-L1161
+// )
+// is not a standard mime type. according to the specs of UPNP
+// http://upnp.org/specs/av/UPnP-av-ConnectionManager-v3-Service.pdf chapter "A.1.1 ProtocolInfo Definition"
+// "The <contentFormat> part for HTTP GET is described by a MIME type RFC https://www.ietf.org/rfc/rfc1341.txt"
+// all the pre-defined "text/*" MIME by IANA is here https://www.iana.org/assignments/media-types/media-types.xhtml#text
+// there is not any subtitle MIME for now (year 2022), we used to use text/srt|ssa|sub|idx, but,
+// kodi support SUP subtitle now, and SUP subtitle is not really a text(see below), to keep it
+// compatible, we suggest only to math the extension
+//
+// main purpose of this array is to shard supported real subtitle formats when kodi act as a UPNP
+// Server or UPNP/DLNA media render
+static const std::string ALLOWED_REAL_SUBTITLE_FORMAT[] = {
```

Comment:
wondering if you can't simply use:

`URIUtils::HasExtension(your_path_or_extension, CServiceBroker::GetFileExtensionProvider().GetSubtitleExtensions());`

Reply:
> 

`CServiceBroker::GetFileExtensionProvider().GetSubtitleExtensions()` [including all  real subtitles + VFS file](https://github.com/xbmc/xbmc/blob/7b8df883185bb8d13d099931ff6a286c13d3a499/xbmc/utils/FileExtensionProvider.cpp#L77)

this is why  they hard coded the subtitle format supported 8 years ago https://github.com/xbmc/xbmc/blame/678e3f80cbbfeecbef56e0b0a3d7db4334920f21/xbmc/network/upnp/UPnPInternal.cpp#L661-L662

if you think a new function like `CServiceBroker::GetFileExtensionProvider().Get**Real**SubtitleExtensions()` is acceptable, I will add it to  FileExtensionProvider() 



Looking forward  to your advise

Reply:
maybe do reuse https://github.com/xbmc/xbmc/blob/be955f9b164f67ca2503392220e948f834a4d92c/xbmc/settings/AdvancedSettings.cpp#L438 but add a blacklist for zip?

Reply:
or separate  `m_subtitlesExtensions` into two variable
one the all the **real subtitle formats** that kodi support
another is currently `m_subtitlesExtensions` - without the  **real subtitle formats** ?

Looking forward to more suggestions.

Reply:
The comment reason is the fact that kodi can later support other subtitle formats, and that array might be updated while for UPnP it'll certainly be forgotten. If you can somehow reuse the same variable but blacklist what should not be supported in UPnP it'd be much better.

But thinking a bit more about this, it won't probably work since the user can rewrite the advanced setting entirely, so it's better to keep the copy here.